File: Xamarin/Xamarin.Native/Xamarin.Native.iOS/Services/NetworkTestService.cs
using CoreFoundation;
using Foundation;
using System;
using System.Collections.Generic;
using System.IO;
using System.Net;
using System.Text;
using UIKit;

namespace Xamarin.Services
{
    public delegate void ImageDownloadEventHandler(UIImage image);

    internal class NetworkTestService
    {
        public event ImageDownloadEventHandler ImageDownloadCompleted;

        public void DownloadImage(string urlString)
        {
            var url = NSUrl.FromString(urlString);
            var data = NSData.FromUrl(url);
            var resultImage = UIImage.LoadFromData(data);
            ImageDownloadCompleted?.Invoke(resultImage);
        }
    }
}

File: Native.iOS/Native.iOS/Services/NetworkTestService.m
//
//  NetworkTestService.m
//  Native.iOS
//
//  Created by Pawel Grzmil on 08/05/16.
//  Copyright © 2016 Pawel Grzmil. All rights reserved.
//

#import "NetworkTestService.h"

@implementation NetworkTestService

- (void)downloadImage:(NSString*)urlString{
    NSURL *url = [NSURL URLWithString:urlString];
    NSData *data = [NSData dataWithContentsOfURL:url];
    UIImage *resultImage = [UIImage imageWithData:data];
    
    if ([self.delegate conformsToProtocol:@protocol(NetworkTestServiceDelegate)]) {
        [self.delegate imageDownloadCompleted:resultImage];
    }
}

@end
